### DISCUSSION TITLE
clang-tidy command line?

### DISCUSSION THREAD

**Discussion by elbrandt** [microsoft/vscode-cpptools#10132] [General]
Is there a way to see the actual commmand line that VSCode is issuing to clang-tidy?

I am trying to figure out why clang-tidy is giving clang-diagnostic-error messages about being unable to find certain header files, though I have configured workspace settings file settings.json to with "C_Cpp.default.includePath" to include the necessary folders, and have configured settings.json to contain "C_Cpp.codeAnalysis.clangTidy.args" entries such as "-Imy_path/to/the/files".

The build (clang and/or gcc and/or aarch64-linux-g++) all build fine with no errors. I am using the cmake extension in VSCode to generate the makefile (Ninja) and run the build.

I can't find the reason why the #include files are not being found by clang-tidy, so it seems that checking the actual command line being issued by VSCode would be a starting point for debugging this.

Thanks in advance for any help debugging this configuration issue.

**Comment by sean-mcmanus** [comment]
Yeah, it's in the C/C++ logging if C_Cpp.loggingLevel is set to "Debug".

Our clang-tidy implementation is dependent on the IntelliSense configuration, such as for the includePath. See https://github.com/microsoft/vscode-cpptools/issues/8874 .
